### PR TITLE
SM-6056: Update Open Data Docs - Activity Notifications

### PR DESCRIPTION
**Pull request by Kellym-Kainos**
1. Updated json schema to handle conditional fields for `object_data` depending on `object_type`. Can be tested here: https://www.jsonschemavalidator.net/
2. Added some missing fields to permit
3. Updated fields to use enum validation where appropriate

### CODE REVIEW

**Review by cpaul92 — APPROVED** [review]

**Review by caoimheb — APPROVED** [review]

File: docs/api-documentation/json/event-notifier-message.json [hunk 1]
```diff
@@ -72,64 +72,154 @@
           },
           "proposed_start_time": {
             "description": "The time the work is proposed to start. Undefined if not provided by the promoter.",
-            "type": "string",
+         	"type": ["string", "null"],
```
Comment (cpaul92) [inline docs/api-documentation/json/event-notifier-message.json:75]: indentation out here?

File: docs/api-documentation/json/event-notifier-message.json [hunk 2]
```diff
@@ -72,64 +72,154 @@
...
           "proposed_end_time": {
             "description": "The time the work is proposed to end. Undefined if not provided by the promoter.",
-            "type": "string",
+            "type": ["string", "null"],
             "format": "date-time"
           },
           "actual_start_date_time": {
             "description": "The date & time the work started",
-            "type": "string",
+        	"type": ["string", "null"],
```
Comment (cpaul92) [inline docs/api-documentation/json/event-notifier-message.json:85]: indentation out here?

File: docs/api-documentation/json/event-notifier-message.json [hunk 3]
```diff
@@ -4,21 +4,13 @@
...
-      "type": "string",
-      "enum": ["WORK_START", "WORK_STOP", "WORK_START_REVERTED", "WORK_STOP_REVERTED", "PERMIT_GRANTED", "PERMIT_ALTERATION_GRANTED", "PERMIT_CANCELLED", "PERMIT_REVOKED", "PERMIT_SUBMITTED", "PERMIT_REFUSED"]
-    },
+  "definitions": {
     "object_data": {
       "description": "The data of the object reference",
       "type": "object",
-      "if": { "object_type": "PERMIT" },
-      "then": {
+        "required": [
```
Comment (caoimheb) [inline docs/api-documentation/json/event-notifier-message.json:11]: I think lines 11 - 208 should all be back one indentation level

**Review by caoimheb — APPROVED** [review]
Approved, pending the few indentation comments

**Pull request merged**